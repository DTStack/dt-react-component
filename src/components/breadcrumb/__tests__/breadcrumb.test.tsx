import React from 'react'
import Breadcrumb from '../index';
import { render, fireEvent, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';
import { createMemoryHistory, Router, Route, IndexRoute } from 'react-router'

const testProps = {
    routes: [
        {
            name: 'home',
            path: '/home'
        },
        {
            name: 'about',
            path: '/about'
        }
    ],
    style: {
        backgroundColor: '#dedede'
    }
}

const App = (props) => {
    return (
        <div>
            {<Breadcrumb {...testProps} />}
            {props.children}
        </div>
    )
}
const About = () => <h1>about page</h1>
const Home = () => <h1>home page</h1>

describe('test breadcrumb', () => {
    afterEach(() => {
        cleanup();
    })
    test('should render correct BreadCrumb with testProps', () => {
        const { getByTestId } = render(<Breadcrumb {...testProps} />);
        const element = getByTestId('test-breadcrumb');
        expect(element).toBeInTheDocument();
        expect(element).toHaveStyle('background-color: #dedede');
    })

    test('should navigate to home router when click ', () => {
        const history = createMemoryHistory()
        const { container, getByTestId } = render(
            <Router history={history}>
                <Route path="/" component={App}>
                    <IndexRoute component={About} />
                    <Route path="/about" component={About} />
                    <Route path="/home" component={Home} />
                </Route>
            </Router>
        );
        expect(container.innerHTML).toMatch('about')
        fireEvent.click(getByTestId('/home-link'))
        expect(container.innerHTML).toMatch('home page')
    })
    test('should not navigate when click', () => {
        const history = createMemoryHistory()
        const { container, getByText } = render(
            <Router history={history}>
                <Route path="/" component={App}>
                    <IndexRoute component={About} />
                    <Route path="/about" component={About} />
                    <Route path="/home" component={Home} />
                </Route>
            </Router>
        );
        fireEvent.click(getByText('about page'));
        expect(container.innerHTML).toMatch('about page');
    })
})
