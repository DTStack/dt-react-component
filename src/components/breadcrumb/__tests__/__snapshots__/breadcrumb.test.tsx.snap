// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test breadcrumb should support BreadCrumb render correct with defaultProps 1`] = `
<div
  class="dtc-breadcrumb"
  style="background-color: rgb(222, 222, 222);"
>
  <nav
    class="ant-breadcrumb"
  >
    <ol>
      <li>
        <span
          class="ant-breadcrumb-link"
        >
          <a
            data-testid="/home-link"
          >
            home
          </a>
        </span>
        <span
          class="ant-breadcrumb-separator"
        >
          <span
            aria-label="right"
            class="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </span>
      </li>
      <li>
        <span
          class="ant-breadcrumb-link"
        >
          about
        </span>
        <span
          class="ant-breadcrumb-separator"
        >
          <span
            aria-label="right"
            class="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </span>
      </li>
    </ol>
  </nav>
</div>
`;
